lib/improver/wind_downscaling.py

```diff
@@ -0,0 +1,967 @@
+            if np.isnan(hld.data).any() or (hld.data == RMDI).any():
+                raise ValueError("height grid contains invalid points")
+            if hld.ndim == 3:
+                try:
+                    xap, yap, zap, _ = self.find_coord_order(hld)
+                    hld.transpose([yap, xap, zap])
+                except Exception:
+                    raise ValueError("height grid different from wind grid")
+            else:
+                try:
+                    hld = next(hld.slices([self.z_name]))
+                except Exception:
+                    raise ValueError("height z coordinate differs from wind z")
+            hld = hld.data
+        return hld
+
+    def check_wind_ancil(self, xwp, ywp):
+        """Check wind vs ancillary file grids.
+
+        Check if wind and ancillary files are on the same grid and if
+        they have the same ordering.
+
+        Parameters:
+        xwp: integer
+            representing the position of the x-axis in the wind cube
+        ywp: integer
+            representing the position of the y-axis of the wind cube
+
+        """
+        xap, yap, _, _ = self.find_coord_order(self.pp_oro)
+        if xwp - ywp != xap-yap:
+            if np.isnan(xap) or np.isnan(yap):
+                raise ValueError("ancillary grid different from wind grid")
+            else:
+                raise ValueError("xy-orientation: ancillary differ from wind")
+
+    def process(self, cube):
+        """Adjust the 4d wind field - cube - (x, y, z including times).
+
+        Parameters
+        ----------
+        cube - iris.cube.Cube
+            The wind cube to be operated upon. Should be wind speed on
+            height_levels for all desired forecast times.
+
+        Returns
+        -------
+        cube
+            The 4d wind field with roughness and height correction
+            applied in the same order as the input cube.
+
+        """
+        wind = cube
+        if not isinstance(wind, iris.cube.Cube):
+            msg = "wind input is not a cube, but {}"
+            raise ValueError(msg.format(type(wind)))
+        (self.x_name, self.y_name, self.z_name,
+         self.t_name) = self.find_coord_names(wind)
+        xwp, ywp, zwp, twp = self.find_coord_order(wind)
+        try:
+            wind.transpose([ywp, xwp, zwp, twp])  # problems with slices
+        except ValueError:
+            wind.transpose([ywp, xwp, zwp])
+        rchc_list = iris.cube.CubeList()
+        if self.l_no_winddownscale:
+            z0_data = None
+        else:
+            z0_data = self.z_0.data
+        roughness_correction = RoughnessCorrectionUtilities(
+            self.a_over_s.data, self.sigma.data, z0_data, self.pp_oro.data,
+            self.model_oro.data, self.ppres, self.modres)
+        self.check_wind_ancil(xwp, ywp)
+        hld = self.find_heightgrid(wind)
+        for time_slice in wind.slices([self.y_name, self.x_name, self.z_name],
```

Comment:
Replaced with slices_over("time"), as I think they're equivalent in  this instance, as the input cube has already been transposed to have a particular ordering to the dimensions.